contracts/extensions/ERC4907A.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: MIT
+// ERC721A Contracts v4.1.0
+// Creator: Chiru Labs
+
+pragma solidity ^0.8.4;
+
+import './IERC4907A.sol';
+import '../ERC721A.sol';
+
+/**
+ * @dev ERC4907 compliant extension of ERC721A.
+ *
+ * The ERC4907 standard https://eips.ethereum.org/EIPS/eip-4907[ERC4907] allows
+ * owners and authorized addresses to add a time-limited role
+ * with restricted permissions to ERC721 tokens.
+ */
+abstract contract ERC4907A is ERC721A, IERC4907A {
+    // The bit position of `expires` in packed user info.
+    uint256 private constant _BITPOS_EXPIRES = 160;
+
+    // Mapping from token ID to user info.
+    //
+    // Bits Layout:
+    // - [0..159]   `user`
+    // - [160..223] `expires`
```

Comment:
As per your setUser and userOf logic bits layout is

// - [96..159] `expires`
// - [160..255] `user`

Reply:
Nope. 

If this is the case, the tests will fail as we use `address(uint160(packed))` to get the address.

Reply:
Solidity Explicit Conversion

```
uint32 a = 0x12345678;
uint16 b = uint16(a);   // b = 0x5678
```

Example : 
packed = 0x000000000000000062D2F4C571C7656EC7AB88B098DEFB751B7401B5F6D8976F

uint160(packed) = 0x71C7656EC7AB88B098DEFB751B7401B5F6D8976F

Reply:
We define the bit at index 0 as the least-significant-bit, and the bit at 255 as the most-significant-bit.